src/Microsoft.Extensions.ApplicationModelDetection/AppModelDetector.cs

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Xml.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.AspNetCore.Hosting
```

Comment:
Why are these in the Microsoft.AspNetCore.Hosting namespace? That seems very at odds with the Microsoft.Extensions.ApplicationModelDetection package name. Why is it named Extensions anyways?